src/index.js

```diff
@@ -4,85 +4,67 @@
-import { IntlProvider, intlShape } from 'react-intl';
+import { createElement } from 'react';
+import { IntlProvider } from 'react-intl';
 import { mount, shallow, render } from 'enzyme';
 
 export const mockIntl = {
   defaultLocale: 'en',
   formatDate: () => 'mock date',
-  formatHTMLMessage: ({ id }) => id,
   formatMessage: ({ id }) => id,
   formatNumber: value => `${value}`,
   formatPlural: value => `${value}`,
   formatTime: value => `${value}`,
-  formatRelative: value => `${value}`,
-  now: () => 0,
+  formatRelativeTime: value => `${value}`,
 };
 
-const messageProxy = new Proxy({}, { get: (_, property) => property }); // eslint-disable-line compat/compat
+const messageProxy = new Proxy({}, {
+  get: (_, property) => property,
+  getOwnPropertyDescriptor: () => ({ configurable: true, enumerable: true }),
+}); // eslint-disable-line compat/compat
+const locale = 'en';
+const defaultLocale = 'en';
 
-// Create the IntlProvider to retrieve context for wrapping around.
-const intlProvider = new IntlProvider({ locale: 'en', messages: messageProxy }, {});
-const { intl } = intlProvider.getChildContext();
-
-const injectIntl = node => cloneElement(node, { intl });
-
-export const shallowContext = { context: { intl } };
-export const mountContext = { context: { intl }, childContextTypes: { intl: intlShape } };
-
-export const mountWithIntl = (node, { context, childContextTypes, ...options } = {}) => {
-  const subject = injectIntl(node);
+export const mountWithIntl = (node, options = {}) => {
   const opts = {
-    context: {
-      ...context,
-      ...mountContext.context,
-    },
-    childContextTypes: {
-      ...childContextTypes,
-      ...mountContext.childContextTypes,
+    wrappingComponent: IntlProvider,
+    wrappingComponentProps: {
+      locale,
+      defaultLocale,
+      messages: messageProxy,
     },
     ...options,
   };
 
-  return mount(subject, opts);
+  return mount(node, opts);
 };
+export const renderWithIntl = (node, options = {}) => {
+  const subject = createElement(IntlProvider, {
+    locale,
+    defaultLocale,
+    messages: messageProxy,
+  }, node);
 
-export const renderWithIntl = (node, { context, childContextTypes, ...options } = {}) => {
-  const subject = injectIntl(node);
-  const opts = {
-    context: {
-      ...context,
-      ...mountContext.context,
-    },
-    childContextTypes: {
-      ...childContextTypes,
-      ...mountContext.childContextTypes,
-    },
-    ...options,
-  };
-
-  return render(subject, opts);
+  return render(subject, options);
 };
 
-export const shallowWithIntl = (node, { context, ...options } = {}) => {
-  const subject = injectIntl(node);
+export const shallowWithIntl = (node, options = {}) => {
   const opts = {
-    context: {
-      ...context,
-      ...shallowContext.context,
+    wrappingComponent: IntlProvider,
+    wrappingComponentProps: {
+      locale,
+      defaultLocale,
+      messages: messageProxy,
     },
     ...options,
```

Comment:
I'm waffling on whether we want to be able to support a user-defined wrappingComponent in addition to the IntlProvider.

```jsx
wrappingComponent:  ({ children }) => {
  const intlProvider = (
    <IntlProvider locale={locale} defaultLocale={defaultLocale} messages={messageProxy}>
      {children}
    </IntlProvider>
  );

  if (options.wrappingComponent) {
    return React.createElement(options.wrappingComponent, options.wrappingComponentProps, intlProvider);
  } else {
    return intlProvider;
  }
},
```

Reply:
Updated in new PR: https://github.com/cerner/terra-toolkit/pull/509